simapp/testutil_network_test.go

```diff
@@ -1,27 +1,79 @@
 package simapp_test
 
 import (
+	"fmt"
+	"math/rand"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/suite"
 
+	sdkmath "cosmossdk.io/math"
 	"cosmossdk.io/simapp"
 
+	cmtcfg "github.com/cometbft/cometbft/config"
+	"github.com/cosmos/cosmos-sdk/client/grpc/cmtservice"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
 	"github.com/cosmos/cosmos-sdk/testutil/network"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	authsim "github.com/cosmos/cosmos-sdk/x/auth/simulation"
+	banksim "github.com/cosmos/cosmos-sdk/x/bank/simulation"
+	govsim "github.com/cosmos/cosmos-sdk/x/gov/simulation"
+	xsim "github.com/cosmos/cosmos-sdk/x/simulation"
+	"github.com/cosmos/go-bip39"
 )
 
 type IntegrationTestSuite struct {
 	suite.Suite
 
+	cfg      network.Config
+	v0app    *simapp.SimApp
+	rand     *rand.Rand
+	accounts []simtypes.Account
+
 	network *network.Network
 }
 
 func (s *IntegrationTestSuite) SetupSuite() {
 	s.T().Log("setting up integration test suite")
 
+	s.rand = rand.New(rand.NewSource(42))
+	params := xsim.RandomParams(s.rand)
+	s.accounts = simtypes.RandomAccounts(s.rand, params.NumKeys())
+	// TODO: improve this hack to get at the bank keeper.
+	s.cfg = network.DefaultConfig(func() network.TestFixture {
+		fixture := simapp.NewTestNetworkFixture()
+		appCtr := fixture.AppConstructor
+		fixture.AppConstructor = func(val network.ValidatorI) servertypes.Application {
+			app := appCtr(val)
+			s.v0app = app.(*simapp.SimApp)
+			return app
+		}
+		return fixture
+	})
+	s.cfg.NumValidators = s.rand.Intn(10)
+	s.cfg.ChainID = fmt.Sprintf("chainid-simapp-%x", s.rand.Uint64())
+	s.cfg.GenesisTime = simtypes.RandTimestamp(s.rand)
+	s.cfg.GenesisTime = time.Unix(0, 0)
+	s.cfg.FuzzConnConfig = cmtcfg.DefaultFuzzConnConfig()
+	s.cfg.FuzzConnConfig.ProbDropConn = .1
+	for i := 0; i < s.cfg.NumValidators; i++ {
+		entropy := make([]byte, 256/8)
+		s.rand.Read(entropy)
```

Comment:
In case even rand.Reader becomes crypto/rand.Reader instead of math/rand.Reader, let's still use n from the length of bytes read
```go
n, _ := s.rand.Read(entropy)
mnemonic, err := bip39.NewMnemonic(entropy[:n])
```